data/client/king_phisher/plugins/data_collection.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import copy
+import datetime
+import hashlib
+import sys
+
+from king_phisher import color
+from king_phisher import its
+from king_phisher import json_ex
+from king_phisher import version
+import king_phisher.client.plugins as plugins
+import king_phisher.client.gui_utilities as gui_utilities
+import king_phisher.client.widget.extras as extras
+
+from smoke_zephyr.utilities import unique
+
+__version__ = '1.0'
+
+def generate_hash(data, algorithm='sha256'):
+	if its.py_v3 and isinstance(data, str):
+		data = data.encode('utf-8')
+	return algorithm + ':' + hashlib.new(algorithm, data).hexdigest()
+
+def generate_signature(data):
+	data = copy.deepcopy(data)
+	data['hash'] = None
+	return generate_hash(json_ex.dumps(data))
+
+def verify_signature(data):
+	if not data.get('hash'):
+		return False
+	return data['hash'] == generate_signature(data)
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Spencer McIntyre', 'Erik Daguerre']
+	title = 'Anonymous Data Collection'
+	description = """
+	By default this plugin will be disabled, You must opt-in to this feature, to
+	have your campaign data anonymize, furthermore you will need to email the data in
+	for analyze.
+
+	By enabling this plugin, you are allowing this plugin to anonymize your campaign
+	data and prompt you to review the anonymous data collected and email it in for processing.
+	"""
+	homepage = 'http://github.com/securestate/king-phisher-plugins'
```

Comment:
Because this will be an official one with the main project, it's home page should just be "https://github.com/securestate/king-phisher". Also note the use of HTTPS.